### PR TITLE
Update usage of TestSink

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Preparing for aspnet/Logging#804

### CODE REVIEW

File: test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs [hunk 1]
```diff
@@ -60,8 +60,8 @@ public void MatchFail_LogsCorrectData()
 
             // Assert
             Assert.Empty(sink.Scopes);
-            Assert.Single(sink.Writes);
-            Assert.Equal(expectedMessage, sink.Writes[0].State?.ToString());
+            var write = Assert.Single(sink.Writes);
+            Assert.Equal(expectedMessage, write.State?.ToString());
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs]: User your new formatted message property.

Comment (JunTaoLuo) [reply test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs]: ugh... I'll have to use the feature branch.

Comment (JunTaoLuo) [reply test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs]: which involves building all the transitive repos that use logging...

Comment (pakrym) [reply test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs]: It's on you conscious then.

Comment (JunTaoLuo) [reply test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs]: I have no conscience.

**Comment by rynowak** [comment]
Can we just stop testing logging instead? #HotTake

**Pull request merged**